Create home page

kravich2 opened the pull request.

kravich2 committed:
home page createion

home page

kravich2 committed:
Finished Home Page v1

finished up the home page

kravich2 committed:
myclub changes

tmohaupt committed:
Update homepage

Update homepage

kravich2 committed:
more changes

kravich2 committed:
Merge branch 'Create_Home_Page' of https://github.com/varangill/ClubHub into Create_Home_Page

varangill commented on UWO.jpeg:
Are UWO.jpeg and UWO2.jpeg still used? If not, they can be removed, and if they are used, they should be in the `client` folder

kravich2 replied:
I removed the photos and have it refer to a URL now.

varangill commented on client/package.json:
```diff
@@ -15,7 +15,7 @@
     "react-bootstrap": "^2.9.1",
     "react-dom": "^18.2.0",
-    "react-router-dom": "^6.19.0"
+    "react-router-dom": "^6.21.1"
```
Any reason for a dependency update or was it an accident? not a big deal right now with how small our codebase is so I'm just curious

kravich2 replied:
it was an accident.

varangill commented on client/src/components/MyClubs.tsx:
```diff
@@ -1,10 +1,117 @@
+  ];
+
+  const [clubs, setClubs] = useState(initialClubs);
+  const [leftClubs, setLeftClubs] = useState([]);
```
From my understanding, `leftClubs` is the list of clubs the user has recently left and we're showing an `undo` option for, right? Not every single club the user has left? If so, this variable name is a bit misleading, renaming it to something like `undoableClubs` would be better

kravich2 replied:
Renamed.

varangill commented on client/src/components/MyClubs.tsx:
```diff
@@ -1,10 +1,117 @@
+  };
+
+  // Function to handle undoing leave action
+  const undoLeave = (clubId) => {
```
Do we really want an undo option for this release? We can discuss it as a group but I think it's something to work on if we have extra time at the end of the year, there's more key functionalities to focus on currently IMO, just leaving a club is fine for now

tmohaupt replied:
I don't see any need for this feature. You can simply just rejoin from the AllClubs page

kravich2 replied:
I thought that if the user accidentally clicked the leave button for the wrong club than it would be easier to have an undo feature instead of trying to search for the club in the AllClubs pages.

varangill replied:
I'd agree with Tyler, don't think we need this tbh

varangill replied:
Gonna say we can just remove the undo stuff from the PR

kravich2 replied:
I removed it

varangill commented on client/src/components/MyClubs.tsx:
```diff
@@ -1,10 +1,117 @@
+
   return (
-    <div>
+    <div style={{ textAlign: 'center' }}>
```
Can you move all in-line style instances like this to a CSS file? @tmohaupt has [an example of moving styles to a CSS file here](https://github.com/varangill/ClubHub/pull/9/commits/7d70e4a6c0551b49dd66e341acd208e4e6e3c6fc) if you need help

There's also in-line styles on lines 45, 49, 61, 65, and the three style constants at the bottom of the file that could be removed to clean up the file a bit

varangill replied:
Still want this done for this PR @kravich2

kravich2 replied:
I entered the changes.

varangill commented on package.json:
```diff
@@ -0,0 +1,5 @@
+{
+  "dependencies": {
```
I assume this was an accident but we don't want `package.json` or `package-lock.json` in the uppermost folder, these can be removed

kravich2 replied:
Removed

varangill reviewed (changes requested):
Looks pretty good so far, few suggestions/comments though

kravich2 committed:
PR Changes

varangill commented on server/env:
```diff
@@ -0,0 +1,5 @@
+USER=postgres
```
This file should be called `.env` (with the dot) and should not be included in the commit, your issues running the server were probably caused by you naming the file `env`

kravich2 replied:
Thanks for telling me, I'll try it now

varangill commented on server/package.json:
```diff
@@ -22,6 +22,7 @@
   "dependencies": {
     "dotenv": "^16.3.1",
     "express": "^4.17.1",
+    "knex": "^3.1.0",
```
You should try to install knex globally instead of adding a dependency for only your machine, if you're having troubles and you can't find a solution online then ask in the group chat for help

varangill replied:
Still want this uninstalled from here and package-lock.json

kravich2 replied:
Uninstalled it

varangill commented on server/src/database/knexfile.js:
```diff
@@ -0,0 +1,13 @@
+module.exports = {
```
We have a knexfile already in Typescript

kravich2 replied:
Removed.

varangill commented on server/src/database/migrations/20240108011223_migration_name.js:
```diff
@@ -0,0 +1,15 @@
+/**
```
Remove this empty migration

kravich2 replied:
Removed.

varangill commented on client/env:
```diff
@@ -0,0 +1 @@
+VITE_BACKEND_URL=localhost:3000
```
Same as other env, should be `.env` and not in the commit

kravich2 replied:
Thanks for telling me, I'll try it out now.

varangill reviewed (changes requested):
More comments

Also it's good practice to commit frequently and base commits on smaller, discrete changes instead of one bigger commit every time you want to push up your changes, for example one small commit from this giant commit could've been "Delete unnecessary files" which deleted the two .jpeg files and package.json and package-lock.json

kravich2 committed:
More Changes from PR

kravich2 committed:
More changes for PR

varangill commented on .gitignore:
```diff
@@ -22,4 +22,6 @@ dist-ssr
\ No newline at end of file
+.env
+server/env
+client/env
```
like I mentioned in the previous review, you shouldn't have any files called `env`, only `.env`, this doesn't need to be in the .gitignore

kravich2 replied:
Deleted the .env files

varangill commented on client/src/components/MyClubs.tsx:
```diff
@@ -1,10 +1,62 @@
+import React, { useState } from 'react';
+import NavBar from './NavigationBar';
+import { Link } from 'react-router-dom';
+import '../MyClubs.css'; // Import the CSS file
```
import this in App.tsx like the other CSS file is imported

kravich2 replied:
added

varangill commented on client/src/components/AllClubs.tsx:
```diff
@@ -1,10 +1,152 @@
+      {/* Total Club Count and Search Bar */}
+      <div
+        style={{
+          position: 'absolute',
```
Lots of styling in this file that should be moved to a new CSS file like the other

kravich2 replied:
Added

varangill reviewed (changes requested):
Still more changes to make - read my review comments carefully please, same thing with unclear commit messages as last time

kravich2 committed:
More PR changes

kravich2 committed:
more PR changes

added MyClubs to App.tsx and added new css file to enter line style from AllClubs.tsx

varangill commented on client/src/components/ClubDetails.tsx:
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
```
What's the purpose of this component? It isn't used anywhere? It looks like it'd serve a similar purpose as `ClubPage` right?

kravich2 replied:
Took it out.

varangill commented on client/src/components/ClubDetails.tsx:
```diff
@@ -0,0 +1,19 @@
+  return (
+    <div>
+      <h2>Club Details Page</h2>
+      <p>Club ID: {clubId}</p>
```
Data object IDs should be for internal use only, they don't need to know the club ID on the UI page

kravich2 replied:
I had the data object id for when I was trying to show the detail description of the club when the user would click the button. I'm still working on the function which is why it is still there.

varangill commented on client/src/components/ClubPage.tsx:
```diff
@@ -1,10 +1,45 @@
 
 export default function ClubPage() {
+  // Get the clubName parameter from the URL
+  const { clubName } = useParams();
```
The club name isn't in parameters, if you want the club name you need to take the ID (which would be in the params) and query the API endpoint for club info. But for now I'd just remove this and use some placeholder club name, this PR is already huge, you can add the API queries in a future PR, it's better to just have smaller PRs

kravich2 replied:
Deleted.

varangill commented on client/src/components/ClubPage.tsx:
```diff
@@ -1,10 +1,45 @@
+          fontFamily: 'Arial, sans-serif'
+        }}
+      >
+        Club page - {clubName ? decodeURIComponent(clubName) : 'Unknown Club'}
```
Weird ChatGPT'd code... Not something we'd need

kravich2 replied:
taken out

varangill commented on client/src/components/ClubPage.tsx:
```diff
@@ -1,10 +1,45 @@
-      Club page
+      {/* Display Club Name */}
+      <h2
+        style={{
```
Styling goes in CSS file, not in-line

kravich2 replied:
Done.

varangill commented on client/src/components/ClubPage.tsx:
```diff
@@ -1,10 +1,45 @@
 }
+
+// Button style for Go Back button
+const buttonStyle = {
```
More styling that should go in a CSS file

varangill commented on client/src/components/MyClubs.tsx:
```diff
@@ -1,10 +1,62 @@
+import React, { useState } from 'react';
+import NavBar from './NavigationBar';
+import { Link } from 'react-router-dom';
+import '../MyClubs.css'; // Import the CSS file
```
You're still importing this CSS file here

varangill commented on client/src/components/MyClubs.tsx:
```diff
@@ -1,10 +1,62 @@
+              <td>
+                {/* Use Link to navigate to ClubPage with the club's ID and name */}
+                <Link
+                  to={`/club/${club.id}?clubName=${encodeURIComponent(club.name)}`}
```
Club name shouldn't be in the URL like this, more weird ChatGPT'd code

varangill commented on client/src/components/MyClubs.tsx:
```diff
@@ -1,10 +1,62 @@
+        placeholder="Search clubs..."
+        value={query}
+        onChange={(e) => setQuery(e.target.value)}
+        className="search-input" // Use CSS class for styling
```
These "Use CSS class for styling" comments are a bit odd, they're not really blockers but they don't serve any purpose? It's known that the `className` is used as a CSS selector, that's how classes and CSS work

varangill commented on client/src/components/AllClubs.tsx:
```diff
@@ -1,10 +1,97 @@
+import React, { useState } from 'react';
+import NavBar from './NavigationBar';
+import { Link } from 'react-router-dom';
+import '../AllClubs.css'; // Import the CSS file
```
Import this in through App.tsx like other CSS files

varangill commented on client/src/components/AllClubs.tsx:
```diff
@@ -1,10 +1,97 @@
+            <tr key={club.id}>
+              <td>
+                <Link
+                  to={`/club/${club.id}?clubName=${encodeURIComponent(
```
Same as my comment in MyClubs.tsx - club name isn't supposed to be like this

varangill commented on client/src/components/AllClubs.tsx:
```diff
@@ -1,10 +1,97 @@
+
+      <table className="clubs-table">
+        <tbody>
+          {filteredClubs.map((club) => (
```
This club listing is remarkably similar to the one in `MyClubs.tsx` and you could just use a common component between the two to get rid of repetitive code, but this is fine for now, it can be changed in a future PR

varangill commented on .gitignore:
```diff
@@ -22,4 +22,6 @@ dist-ssr
\ No newline at end of file
+.env
+server/env
+client/env
```
Delete these `env` files from the .gitignore too please

varangill commented on client/src/components/AllClubs.tsx:
```diff
@@ -1,10 +1,97 @@
+              </td>
+            </tr>
+          ))}
+          {joinedClubs.map((club) => (
```
With the way this is formatted, you're going to have every possible club listed out first (including ones you're in), then you'll have every club you're in, so you'll have duplicate clubs and users will see a "Join" button on clubs they're already in.

IMO you can just remove join buttons from this AllClubs page and move it to the actual specific club page, so users have to click onto a club's page before joining it. Then you can get rid of `joinedClubs`. It'll be a lot cleaner and an easier fix to the above issue

varangill reviewed (changes requested):
A lot of the stuff in this review, particularly the CSS stuff, is stuff I mentioned in a previous review on this exact same PR. Please try to keep previous feedback in mind when coding

kravich2 committed:
PR Changes

added in the css file for Clubpage.

kravich2 committed:
Implemented Join Club in page and changed all clubs page

varangill committed:
Clean up code

varangill committed:
Refactor club list into common component
